test/test_util.jl

```diff
@@ -20,9 +20,57 @@ macro trixi_testset(name, expr)
         using Test
         using TrixiBase
 
+        # We also include this file again to provide the definition of
+        # the other testing macros. This allows to use `@trixi_testset`
+        # in a nested fashion and also call `@test_nowarn_mod` from
+        # there.
+        include(@__FILE__)
+
         @testset verbose=true $name $expr
         end
 
         nothing
     end
 end
+
+"""
+    @test_nowarn_mod expr
+
+Modified version of `@test_nowarn expr` that prints the content of `stderr` when
+it is not empty and ignores some common info statements printed in Trixi.jl
+uses.
+"""
+macro test_nowarn_mod(expr, additional_ignore_content=String[])
+    quote
+        let fname = tempname()
+            try
+                ret = open(fname, "w") do f
+                    redirect_stderr(f) do
+                        $(esc(expr))
+                    end
+                end
+                stderr_content = read(fname, String)
+                if !isempty(stderr_content)
+                    println("Content of `stderr`:\n", stderr_content)
+                end
+
+                # Patterns matching the following ones will be ignored. Additional patterns
+                # passed as arguments can also be regular expressions, so we just use the
+                # type `Any` for `ignore_content`.
+                ignore_content = Any["[ Info: You just called `trixi_include`. Julia may now compile the code, please be patient.\n"]
+                append!(ignore_content, $additional_ignore_content)
+                for pattern in ignore_content
+                    stderr_content = replace(stderr_content, pattern => "")
+                end
+
+                # We also ignore simple module redefinitions for convenience. Thus, we
+                # check whether every line of `stderr_content` is of the form of a
+                # module replacement warning.
+                @test occursin(r"^(WARNING: replacing module .+\.\n)*$", stderr_content)
+                ret
+            finally
+                rm(fname, force=true)
```

Comment:
I just noticed - this is a different style than in Trixi.jl (`force = true`). Is this a deliberate deviation from SciML, @efaulhaber?

Reply:
I copied this from TrixiParticles, where we did not change the style to the one with the whitespaces when SciML did and instead stayed with the more common Julia style.

Reply:
https://github.com/trixi-framework/TrixiBase.jl/pull/7

Reply:
OK, let's move the discussion to #7 then.

Reply:
I'm not sure whether it's more common to have spaces around kwargs or not 🤷